### ISSUE TITLE
soft98.ir [Anti-Adblock]

### ISSUE THREAD

**Issue by nimasaj** [uBlockOrigin/uAssets#3011]
### URL(s) where the issue occurs

`https://soft98.ir`
`https://soft98.ir/software/office/365-notepad.html`

### Describe the issue

Anti adblock popup
Also, it disables the links

### Screenshot(s)
```
1. https://cdn.pbrd.co/images/HvR7w52.png 
Popup showing you have adblock turned on and asks you to disable it.


2. https://cdn.pbrd.co/images/HvRvgUd.png
A picture of download section of second URL (links inside the page are not working as long as it detects adblock)


3. https://cdn.pbrd.co/images/HvRvlOG.png
A picture of download section of 2nd URL in case it detect javascript is blocked. (all links are gone and it asks you to turn on javascript)

```

### Versions

- Browser/version: Chrome 67.0.3396.99
- uBlock Origin version: 1.16.14

### Settings

using following filter list which blocks `##*[id^="adx"]`, `##*[class^="adx"]`, `##a[href*="&utm_medium=banner&utm_campaign="]>img` and `##*[href*="account.zula.ir/pages/new-iran?utm_source"]>img`

Filter list: `http://mynext.pro/BLF.txt`

### Notes

Issue is loaded from `soft98.ir/templates/soft98_default/js/jquery.min.js` and mainly following part:

```
    function(e) {
        ! function(t) {
            var n = $(".adx");
            n.on("contextmenu mousedown", function(e) {
                e.preventDefault()
            });
            var i = 'جهت مشاهده لینک، افزونه حذف تبلیغات را غیرفعال و بعد <a class="tooltip-link" href="#" onclick="location.reload();return false;">بارگذاری مجدد</a> کنید.';
            n.each(function() {
                if ($(this).is(":hidden")) return t(i, e.length.long, "lg").show(), $("#main").find(".download-list-link, .card-title-link, .card-footer .btn-success").each(function() {
                    var e = $(this);
                    e.attr("href", "#"), e.click(function(e) {
                        e.preventDefault()
                    }), e.attr({
                        title: i,
                        "data-toggle": "tooltip",
                        "data-size": "lg",
                        "data-trigger": "click",
                        "data-html": !0,
                        "data-placement": "left"
                    })
                }), !1
            })
        }(e.toast)
    }(application),
```

I tried with `soft98.ir##script:inject(abort-on-property-read.js, $, n)` but it shows picture 3, linked above.

**Comment by okiehsch** [comment]
I am sorry, but I have no clue what I am looking at, I can't read Farsi and all I can tell you is that the 
`Adblock-Iran` list is causing the issue, without it I don't get the "adblock-message" and the "stuff" in your screenshot appears on my end.

You should really report it at https://github.com/farrokhi/adblock-iran.

**Comment by okiehsch** [comment]
If I enable `Adblock-Iran`  I have to allow
`soft98.ir/templates/soft98_default/js/jquery.min.js` and 
disable cosmetic filtering for the site, I then have no obvious issue.

**Comment by okiehsch** [comment]
If II enable `http://mynext.pro/BLF.txt`

````
@@||soft98.ir^$generichide
soft98.ir##.adx-link
````
seems to work on my end.

**Comment by nimasaj** [comment]
Yes, disabling cosmetic filtering solves the issue, but allows ads to be seen there. Isn't there any other way to disable Anti-Adblock on this domain (script injection)?

I tried with Adblock-Iran only (and disabling `mynext.pro/BLF.txt`) and disabling cosmetic filtering, still there is a problem as picture 3 is showing.

With your solution on 4th comment, ads will be visible again on main page (top of the page and also right pane)

**Comment by okiehsch** [comment]
>  tried with Adblock-Iran only (and disabling mynext.pro/BLF.txt) and disabling cosmetic filtering, still there is a problem as picture 3 is showing.

You have to allow `soft98.ir/templates/soft98_default/js/jquery.min.js` the Adblock-Iran list has the filter
`||soft98.ir/templates/soft98_default/js/$script` and that breaks the site.

> ads will be visible again on main page

Can you post a screenshot of the ads you see with `mynext.pro/BLF.txt` enabled and the filters
````
@@||soft98.ir^$generichide
soft98.ir##.adx-image
soft98.ir##.adx-link
````
added to your filter list.

**Comment by nimasaj** [comment]
This is an image with your proposed filters:
`https://cdn.pbrd.co/images/HvRKqAh.png`

It could be fixed with following filters, but isn't there any way to make a script injection rule for such cases?
```
soft98.ir#@#*[id^="adx"]
soft98.ir#@#*[class^="adx"]
soft98.ir##div[class^="adx-"]
soft98.ir##.adx-image
soft98.ir##.adx-link
soft98.ir##.adx__inner
```

**Comment by okiehsch** [comment]
The detection is inside `soft98.ir/templates/soft98_default/js/jquery.min.js` any aborted function will in all likelyhood break the script and lead to your third screenshot `https://cdn.pbrd.co/images/HvRvlOG.png`.

**Comment by nimasaj** [comment]
Issue is back, so I open this topic again.
A script on website is checking for ".adx1". `var t = $(".adx1, .adx1 *");`
I have fixed most of the issue by following filters, in my list `http://mynext.pro/uBOPa.txt`. 
```
soft98.ir##div[class="row"]:if(div>div[class="col-lg pl-0 pr-0 pl-sm-3 pr-sm-3 pr-sm-3"])
soft98.ir##div[class="row"]:if(div>div[class="col-lg pt-3 pl-0 pr-0 pl-sm-3 pr-sm-3 pt-lg-0 pr-lg-0"])
soft98.ir##div[class="col-12 col-lg"]:if(div>div[class^="adx"])
soft98.ir##section[class="card d-block bg-light"]:if(section>header>h4:has-text(پشتیبان سرور سافت))
soft98.ir##section[class="card bg-light"]:if(section>header>h4:has-text(مکان تبلیغات))
```
but these are temporary fixed and still ads in other pages of website could not be blocked with cosmetic filtering. A sample link is: `https://soft98.ir/software/pdf/2165-iceni-technology-infixpro-pdf-editor.html`

This image is showing the issue: `https://cdn.pbrd.co/images/HCcdGBr.png`

Script is loaded from `soft98.ir/templates/soft98_default/js/jquery.min.js`, and a copy of decrypted script is available at `http://pasted.co/de858c1a`.

**Comment by nimasaj** [comment]
It stops the nag, but links (in all pages, including homepage) will be disabled. Have a look at these images please.
```
https://cdn.pbrd.co/images/HCcX0yf.png
https://cdn.pbrd.co/images/HCcX528.png
```

**Comment by mapx-** [comment]
usig only default setup + iran list + my the filters above I get no issues

**Comment by nimasaj** [comment]
I updated my last comment to put new images, please have a look at them again. Iran list has removed filters related to this website. Although to make sure I tried again, and links are disabled. Please check and compare the links with and without adblocker.

**Comment by okiehsch** [comment]
````
@@||soft98.ir^$generichide
soft98.ir##div[class="row"]:if(div>div[class="col-lg pl-0 pr-0 pl-sm-3 pr-sm-3 pr-sm-3"])
soft98.ir##div[class="row"]:if(div>div[class="col-lg pt-3 pl-0 pr-0 pl-sm-3 pr-sm-3 pt-lg-0 pr-lg-0"])
soft98.ir##div[class="col-12 col-lg"]:if(div>div[class^="adx"])
soft98.ir##section[class="card d-block bg-light"]:if(section>header>h4:has-text(پشتیبان سرور سافت))
soft98.ir##section[class="card bg-light"]:if(section>header>h4:has-text(مکان تبلیغات))
soft98.ir##.adx-image:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
````

works on my end with a defaut setup.

**Comment by mapx-** [comment]
Does this 1 work for you ?
`||ironia.ml/$image,domain=soft98.ir,redirect=3x2-transparent.png`

**Comment by mapx-** [comment]
What if you use only
`||ironia.ml/$image,domain=soft98.ir,redirect=3x2-transparent.png`

**Comment by nimasaj** [comment]
yes, but it leaves a big ad-placeholder left, as shown in this image, `https://cdn.pbrd.co/images/HCdelFi.png`

**Comment by nimasaj** [comment]
Filter by @okiehsch fixed the issue. I'm closing this topic. Thanks to both of you for helping.

**Comment by okiehsch** [comment]
@nimasaj just fyi,
like I said, there is high probabilty of site breakage if you try to disable part of
`soft98.ir/templates/soft98_default/js/jquery.min.js` with a `script-injection` filter.

**Comment by nimasaj** [comment]
@okiehsch , if this script is written this much hard to break and invulnerable . Then we would see more and more websites using similar methods to protect themselves from Adblockers.

**Comment by nimasaj** [comment]
Issue is back again. Thus I open this topic if anyone could come up with a solution.
 
<details>
  <summary>Click to expand related section.</summary>
.

URL: `https://soft98.ir/templates/soft98_default/js/jquery.min.js`

They changed **var t** value.
```
    function(e) {
        $($("html")[0].shadowRoot).find("style").remove(), $("style:contains(important)").remove();
        var t = $(".adx, .adx > *");
        t.on("contextmenu mousedown", function(e) {
                e.preventDefault()
            }),
            function n() {
                try {
                    if (e.viewport.is(">= md")) {
                        var i = 'جهت مشاهده لینک، افزونه حذف تبلیغات را در مرورگر خود غیرفعال و <a class="tooltip-link" href="#" onclick="location.reload();return false;">صفحه را دوباره بارگذاری</a> کنید.';
                        t.each(function() {
                            var e = $(this),
                                t = e.attr("style");
                            if (e.is(":hidden")) throw i;
                            var n = e.offset().left;
                            if (n < 0 || n > $(window).width()) throw i;
                            if (-1 !== e.css("visibility").indexOf("hidden")) throw i;
                            if (e.width() < 1) throw i;
                            if (e.height() < 1) throw i;
                            if (t && -1 !== t.indexOf("adguard")) throw i;
                            if (!$.trim(e.html())) throw i
                        })
                    }
                    setTimeout(n, 3e3)
                } catch (t) {
                    e.toast(t, e.length.long, "lg").show(), $("#main").find(".download-list-link, .card-title-link, .card-footer .btn-success").attr("href", "#").click(function(e) {
                        e.preventDefault()
                    }).attr({
                        title: t,
                        "data-toggle": "tooltip",
                        "data-size": "lg",
                        "data-trigger": "click",
                        "data-html": !0,
                        "data-placement": "left"
                    })
                }
            }()
    }(application),
```
</details>

.

<details>
  <summary>Filters which were working previously, but not anymore:</summary>

```
||kaprila.com*/show.*$third-party
soft98.ir##div[class="row"]:if(div>div[class="col-lg pl-0 pr-0 pl-sm-3 pr-sm-3 pr-sm-3"])
soft98.ir##div[class="row"]:if(div>div[class="col-lg pt-3 pl-0 pr-0 pl-sm-3 pr-sm-3 pt-lg-0 pr-lg-0"])
soft98.ir##section[class="card d-block bg-light"]:if(section>header>h4:has-text(پشتیبان سرور سافت))
soft98.ir##section[class="card bg-light"]:if(section>header>h4:has-text(مکان تبلیغات))
soft98.ir##.adx-image:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
soft98.ir##div[class="col-12 col-lg"]:if(div>div[class^="ad2"])
soft98.ir##div[class="col-12 col-lg"]:if(div>div[class^="acdb"])
soft98.ir##div[class="col-12 col-lg"]:if(div>div[class^="adx"])
soft98.ir##div[class="col-12 col-lg"]:if(div>div[class^="adzb"])
soft98.ir##div[class="col-12 col-lg"]:if(div>div>a[title*="زولا"])
soft98.ir##div[class="col-12 col-lg"]:if(div>div>a[href*="/bit.ly/"])
soft98.ir##div[class="container ad2b"]:if(div>div[class^="ad2"])
soft98.ir##div[class="container acdb"]:if(div>div[class^="acdb"])
soft98.ir##div[class="container acdb"]:if(div>div>a[title*="سونی"])
soft98.ir##div[class="container adzb"]:if(div>div[class^="adzb"])
soft98.ir##a[href*="/bit.ly/"]:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
soft98.ir##a[href*="/clc.to/"]:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
soft98.ir##a[href*="/dfile.ir/make/"]:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
soft98.ir##*[class^="acdb "]:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
soft98.ir##*[class^="ad2b "]:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
soft98.ir##*[class^="adzb "]:style(height: 0px !important; width: 0px !important; visibility: collapse !important;)
```
</details>
.

Google Chrome Version 70.0.3538.110 (Official Build) (64-bit)
uBlock Origin v1.17.4
.

**Comment by okiehsch** [comment]
What about 
````
@@||soft98.ir^$generichide
||kaprila.com*/show.*$third-party
soft98.ir##.adx-image:style(height: 1px !important; width: 1px !important; visibility: collapse !important;)
soft98.ir##.adx-images:style(height: 1px !important; width: 1px !important; visibility: collapse !important;)
````

**Comment by nimasaj** [comment]
You're right. Based on your suggestion, the following ones are working.
```
||kaprila.com*/show.*$third-party
soft98.ir##*[class="col-lg pl-0 pr-0 pl-sm-3 pr-sm-3 pr-sm-3"]
soft98.ir##*[class^="adx-image"]:style(height: 1px !important; width: 1px !important; visibility: collapse !important;)
soft98.ir##*[class="col-lg pt-3 pl-0 pr-0 pl-sm-3 pr-sm-3 pt-lg-0 pr-lg-0"]:style(height: 1px !important; width: 1px !important; visibility: collapse !important;)
soft98.ir##*[class="card d-block bg-light"]:style(height: 1px !important; width: 1px !important; visibility: collapse !important;)
```
Though, I couldn't make  id="sidebar_sticky" hidden, shown in following image, but that one is not that important if there is no easy fix for it.
`https://cdn.pbrd.co/images/HTSEIj0.png`

Thanks @okiehsch

**Comment by smed79** [comment]
@SlashArash @farrokhi fyi.

**Comment by rh1363** [comment]
WoW!!!
The Problem is back again!
All the Adds Coming Back again!
https://soft98.ir/
Please Update the blocking List.

**Comment by mapx-** [comment]
Be sure you are using this list:
https://github.com/farrokhi/adblock-iran

**Comment by nimasaj** [comment]
@rh1363, New ads are covered. Please update your list and check again.

**Comment by mapx-** [comment]
@rh1363  and maybe you should use uBo

**Comment by rh1363** [comment]
thank you
the adds is disappeared. but, the scripts that hidden links is still active!

**Comment by nimasaj** [comment]
@rh1363 check again please. By the way, there is no hidden link for me. In case you're using some other Farsi lists, un-mark them and check again. I do not get any hidden link with my list + default lists on uBO.

**Comment by rh1363** [comment]
I Update the List, but the problem change again and I saw some adds!
https://pasteboard.co/HUVKfmR.jpg
would you please revise the list blocking codes of soft98.ir?
there is a suspicious problem!

**Comment by mapx-** [comment]
@rh1363  are you using uBo or ABP ? remove ABP and **keep only uBo** then test again

**Comment by mapx-** [comment]
ok., I see you are using uBo now

**Comment by rh1363** [comment]
I'm Using uBlock Origin and uBOPa filter list.
WoW!
Surprisingly the add:
https://pasteboard.co/HUVKfmR.jpg
coming back again!!!
why this crazy things happen?!

**Comment by rh1363** [comment]
Hi
I Update uBOPa filter list Just now.
Surprisingly, the script of soft98.ir is active and the whole site is disabled!
Please Update....

**Comment by mapx-** [comment]
What list is this one: `uBOPa` => it's `IRN: Adblock-Iran` or something else ? address ?

**Comment by nimasaj** [comment]
@mapx- Either he is the maintainer of that website, or website developer is monitoring this topic. Everytime I made a change in filter list, website changed immediately. It doesn't make sense to make any more changes, at least for now.

**Comment by farrokhi** [comment]
I had the same experience that someone was reporting this to me and as soon as I released an updated and asked him to test, they change the website in like 10 minutes to render the filterlist ineffective. I had the feeling that the reported was the owner or developer of the website.
I suggest we find an alternative method to block the ads on this specific site.

**Comment by rh1363** [comment]
> 
> 
> What list is this one: `uBOPa` => it's `IRN: Adblock-Iran` or something else ? address ?

https://cdn1.imggmi.com/uploads/2019/1/5/537452959c6a05df927b5dbf96a7b0e0-full.jpg

**Comment by rh1363** [comment]
> 
> 
> @mapx- Either he is the maintainer of that website, or website developer is monitoring this topic. Everytime I made a change in filter list, website changed immediately. It doesn't make sense to make any more changes to fix that website, at least for now.
> 
> He was using my filter list (uBOPa) on this address: `soft98.ir`

Thus
would you please instruct me how to identify script and block it?
and also instruct how to find correct add line code for disabling adds.
---------
You make good and throw it into the river, then God will back you at the desert! (a well known Iranian saying! :D)

**Comment by rh1363** [comment]
> 
> 
> I had the same experience that someone was reporting this to me and as soon as I released an updated and asked him to test, they change the website in like 10 minutes to render the filterlist ineffective. I had the feeling that the reported was the owner or developer of the website.
> I suggest we find an alternative method to block the ads on this specific site.

WoW!
Believe me I'm not the developer of soft98!
I'm a simple worker! :D 
.
You right.
we have to find new method.
I hate people that want to feeeeeed innocent users the damn advertisements!

**Comment by sadeghimahdi7474** [comment]
کماکان تبلیغات سایت سافت 98 با udblock origin پاک نمیشه و من از اسکریپت زیر برای حذف تبلیغات استفاده کردم به نظرم سازندگان این اسکریپت رو در پلاگین جای بدن:

https://github.com/DRSDavidSoft/soft98-pro

موضوع دیگه اینکه یوتیوب ویدیو ها رو با این پلاگین حذف تبلیغات از چند روز اخیر باز نمیکنه و باید پلاگین رو غیر فعالش کرد.

**Comment by sadeghimahdi7474** [comment]
The ads on the Soft98 website are still not being blocked by uBlock Origin. I’ve been using the following script to remove the ads, and I think the developers should consider integrating this script into the plugin:

https://github.com/DRSDavidSoft/soft98-pro

Another issue is that, for the past few days, YouTube videos have not been loading when this ad-blocking plugin is enabled. The videos only work after disabling the plugin.
